src/native/libs/CMakeLists.txt

```diff
@@ -144,6 +144,7 @@ if (CLR_CMAKE_TARGET_UNIX OR CLR_CMAKE_TARGET_BROWSER)
         add_compile_options(-Wno-typedef-redefinition)
         add_compile_options(-Wno-c11-extensions)
         add_compile_options(-Wno-thread-safety-analysis)
+        add_compile_options(-Wno-strict-prototypes)
```

Comment:
is this about zlib?

Reply:
> If it's about zlib, I ran into something like this with trying to bump us to 1.2.12. Ended up adding a function `set_zlib_source_files_properties` to `src/native/external/zlib.cmake` to turn off strict prototypes just for those source files: [3c3d19b](https://github.com/dotnet/runtime/commit/3c3d19b06532f565c617dae18d7fb23e9eafdab5)

Part of the warnings come indeed from zlib IIRC, there are many more in other parts as well.

I have opened https://github.com/dotnet/runtime/issues/69142 to not forget to fix them.